### PR TITLE
Create namespace patch test

### LINKED ISSUE #86051: Write patchCoreV1Namespace Mock Test for Conformance - +1 Coverage
This issue proposes the writing of a new test for the `patchCoreV1Namespace` API endpoint with the intention of promoting it to Conformance. This issue contains a very masic **mock** test in order to start the conversation about what a good e2e test would look like for this endpoint. The code below is verified to be hitting the intended endpoint, show as per the queries in APIsnoop's live view of the cluster's audit logs.

## Our Kubectl Apply

```shell
kubectl apply -f "https://raw.githubusercontent.com/cncf/apisnoop/master/deployment/k8s/raiinbow.yaml" --namespace=kube-system
```

# Identify a Feature Using APISnoop

According to this query, there is only one API endpoint which is for Namespace that isn't being covered.

```sql-mode
SELECT                                
  operation_id,                       
  k8s_action,                         
  path,                               
  description                         
  FROM untested_stable_core_endpoints 
  where operation_id like '%Namespace'
 ORDER BY operation_id desc           
       ;                              
```

```sql-mode
     operation_id     | k8s_action |           path            |               description                
----------------------+------------+---------------------------+------------------------------------------
 patchCoreV1Namespace | patch      | /api/v1/namespaces/{name} | partially update the specified Namespace 
(1 row)                                                                                                   
```

We selected **patchCoreV1Namespace** to write a test for.

## Example Test in Go
Dependencies:
```shell
go get -v -u k8s.io/apimachinery/pkg/apis/meta/v1
go get -v -u k8s.io/client-go/kubernetes
go get -v -u k8s.io/client-go/tools/clientcmd
```

The mock test:
```go
package main

import (
  "fmt"
  "flag"
  "os"
  metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
  "k8s.io/client-go/kubernetes"
  "k8s.io/client-go/tools/clientcmd"
)

func main() {
     // uses the current context in kubeconfig
     kubeconfig := flag.String("kubeconfig",
       fmt.Sprintf("%v/%v/%v", os.Getenv("HOME"), ".kube", "config"),
       "(optional) absolute path to the kubeconfig file")
     flag.Parse()
     config, err := clientcmd.BuildConfigFromFlags("", *kubeconfig)
     if err != nil {
       fmt.Println(err)
     }
     // make our work easier to find in the audit_event queries
     config.UserAgent = "live-test-writing"
     // creates the clientset
     clientset, _ := kubernetes.NewForConfig(config)
     // patch namespace labels
     _, err = clientset.CoreV1().Namespaces().Patch("default", "MergePatchType", []byte(`{"metadata":{"name":"default","labels":{"testProperty":"testValue"}}}`))

     if err != nil {
       fmt.Println(err)
       return
     }

     ns, err := clientset.CoreV1().Namespaces().List(metav1.ListOptions{
       LabelSelector: "testProperty=testValue",
     })

     if err != nil {
       fmt.Println(err)
       return
     }

     if ns.Items[0].ObjectMeta.Name != "default" {
       fmt.Println("[status] failed to find patched Namespace")
       return
     }
     fmt.Println("[status] updated namespace sucessfully")
}
```

# Verify with APISnoop

```sql-mode
select * from endpoints_hit_by_new_test where useragent like 'live%';
```

```sql-mode
     useragent     |      operation_id      | hit_by_ete | hit_by_new_test 
-------------------|------------------------|------------|-----------------
 live-test-writing | listCoreV1Namespace    |         16 |               1
 live-test-writing | patchCoreV1Namespace   |          0 |               2
(3 rows)

```

```sql-mode
select * from projected_change_in_coverage;
```

```sql-mode
   category    | total_endpoints | old_coverage | new_coverage | change_in_number 
---------------|-----------------|--------------|--------------|------------------
 test_coverage |             438 |          183 |          184 |                1
(1 row)

```

# Final notes
If a test with these calls is build and merged, conformance will go up by 1 endpoint.

### PR DESCRIPTION
**Pull request by BobyMCbobs**
**What type of PR is this?**
/kind cleanup

**What this PR does / why we need it**:
This PR adds a test for the untested endpoint patchCoreV1Namespace.

**Which issue(s) this PR fixes**:
Fixes #86051

**Special notes for your reviewer**:
Adds +1 endpoint coverage (good for conformance)

**Does this PR introduce a user-facing change?**:
```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:
```docs
NONE
```

/area testing
/sig testing

### CODE REVIEW

File: test/e2e/apimachinery/namespace.go [hunk 1]
```diff
@@ -246,4 +248,38 @@ var _ = SIGDescribe("Namespaces [Serial]", func() {
...
+		namespaceName := "nspatchtest"
+		_, err := f.ClientSet.CoreV1().Namespaces().Create(&v1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: namespaceName,
+			},
+		})
+		framework.ExpectNoError(err, "failed creating Namespace")
+
+		ginkgo.By("patching the Namespace")
+		nspatch := fmt.Sprintf(`{"metadata":{"labels":{"testProperty":"testValue"}}}`, namespaceName)
```
Comment (neolit123) [inline test/e2e/apimachinery/namespace.go]: `testProperty` -> `testLabel`

File: test/e2e/apimachinery/namespace.go [hunk 2]
```diff
@@ -246,4 +248,38 @@ var _ = SIGDescribe("Namespaces [Serial]", func() {
...
+		_, err = f.ClientSet.CoreV1().Namespaces().Patch(namespaceName, types.StrategicMergePatchType, []byte(nspatch))
+		framework.ExpectNoError(err, "failed to patch Namespace")
+
+		ginkgo.By("finding the namespace with it's LabelSelector")
```
Comment (neolit123) [inline test/e2e/apimachinery/namespace.go]: `it's` -> `its`

File: test/e2e/apimachinery/namespace.go [hunk 3]
```diff
@@ -246,4 +248,38 @@ var _ = SIGDescribe("Namespaces [Serial]", func() {
...
+		namespaces, err := f.ClientSet.CoreV1().Namespaces().List(metav1.ListOptions{
+			LabelSelector: "testProperty=testValue",
```
Comment (neolit123) [inline test/e2e/apimachinery/namespace.go]: [1]

File: test/e2e/apimachinery/namespace.go [hunk 4]
```diff
@@ -246,4 +248,38 @@ var _ = SIGDescribe("Namespaces [Serial]", func() {
...
+		})
+		framework.ExpectNoError(err, "failed creating Namespace")
+
+		ginkgo.By("patching the Namespace")
+		nspatch := fmt.Sprintf(`{"metadata":{"labels":{"testProperty":"testValue"}}}`, namespaceName)
+		_, err = f.ClientSet.CoreV1().Namespaces().Patch(namespaceName, types.StrategicMergePatchType, []byte(nspatch))
+		framework.ExpectNoError(err, "failed to patch Namespace")
+
+		ginkgo.By("finding the namespace with it's LabelSelector")
+		namespaces, err := f.ClientSet.CoreV1().Namespaces().List(metav1.ListOptions{
```
Comment (neolit123) [inline test/e2e/apimachinery/namespace.go]: listing feels like a separate test, from patching.

a good way to verify if the namespace is patched would be to `Get` it and check if it has the label.

File: test/e2e/apimachinery/namespace.go (hunk 1 above)
Comment (BobyMCbobs) [reply test/e2e/apimachinery/namespace.go]: I can update this, but I thought that it makes sense as properties have values in JSON.

Comment (neolit123) [reply test/e2e/apimachinery/namespace.go]: my vote goes for `testLabel`.

File: test/e2e/apimachinery/namespace.go (hunk 4 above)
Comment (BobyMCbobs) [reply test/e2e/apimachinery/namespace.go]: Either listing or getting will achieve the same thing, listing will scale better but for this I've updated to your request in f13b448bc2db1

File: test/e2e/apimachinery/namespace.go (hunk 2 above)
Comment (BobyMCbobs) [reply test/e2e/apimachinery/namespace.go]: Thanks, I've update this in f13b448bc2db1

File: test/e2e/apimachinery/namespace.go [hunk 5]
```diff
@@ -246,4 +248,29 @@ var _ = SIGDescribe("Namespaces [Serial]", func() {
...
+			},
+		})
+		framework.ExpectNoError(err, "failed creating Namespace")
+
+		ginkgo.By("patching the Namespace")
+		nspatch := fmt.Sprintf(`{"metadata":{"labels":{"testProperty":"testValue"}}}`, namespaceName)
+		_, err = f.ClientSet.CoreV1().Namespaces().Patch(namespaceName, types.StrategicMergePatchType, []byte(nspatch))
+		framework.ExpectNoError(err, "failed to patch Namespace")
+
+		ginkgo.By("get the namespace and ensuring it has a LabelSelector")
```
Comment (neolit123) [inline test/e2e/apimachinery/namespace.go]: "a LabelSelector" should be just "the label".

**Comment by neolit123** [comment]
please keep the commits squashed to 1 - i.e. amend, force push.

File: test/e2e/apimachinery/namespace.go (hunk 1 above)
Comment (BobyMCbobs) [reply test/e2e/apimachinery/namespace.go]: I've updated it in https://github.com/kubernetes/kubernetes/pull/86854/commits/8abdc97a53d768394613cd89e467d4ac8b8d8c9a

File: test/e2e/apimachinery/namespace.go (hunk 5 above)
Comment (BobyMCbobs) [reply test/e2e/apimachinery/namespace.go]: Updated in https://github.com/kubernetes/kubernetes/pull/86854/commits/8abdc97a53d768394613cd89e467d4ac8b8d8c9a

File: test/e2e/apimachinery/namespace.go [hunk 6]
```diff
@@ -246,4 +248,29 @@ var _ = SIGDescribe("Namespaces [Serial]", func() {
...
+			},
+		})
+		framework.ExpectNoError(err, "failed creating Namespace")
+
+		ginkgo.By("patching the Namespace")
+		nspatch := fmt.Sprintf(`{"metadata":{"labels":{"testLabel":"testValue"}}}`, namespaceName)
+		_, err = f.ClientSet.CoreV1().Namespaces().Patch(namespaceName, types.StrategicMergePatchType, []byte(nspatch))
+		framework.ExpectNoError(err, "failed to patch Namespace")
+
+		ginkgo.By("get the namespace and ensuring it has the label")
```
Comment (neolit123) [inline test/e2e/apimachinery/namespace.go]: final minor nit: missed that namespace is lowercase here, which is inconsistent with the rest of the `.By` calls.

Comment (neolit123) [reply test/e2e/apimachinery/namespace.go]: on the other hand existing code has it lowercase everywhere:
https://github.com/kubernetes/kubernetes/pull/86854/files#diff-35baea3ee2d40a29c405b9114bde6c84R189
so probably better to lowercase it in this PR too.

Comment (BobyMCbobs) [reply test/e2e/apimachinery/namespace.go]: Updated in https://github.com/kubernetes/kubernetes/pull/86854/commits/554eb47e512aedfbf511d621a2584ce5f3559776

**Review by neolit123 — COMMENTED** [review]
/lgtm

File: test/e2e/apimachinery/namespace.go [hunk 7]
```diff
@@ -246,4 +248,29 @@ var _ = SIGDescribe("Namespaces [Serial]", func() {
...
+		namespaceName := "nspatchtest"
+		_, err := f.ClientSet.CoreV1().Namespaces().Create(&v1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: namespaceName,
+			},
+		})
+		framework.ExpectNoError(err, "failed creating Namespace")
+
+		ginkgo.By("patching the Namespace")
+		nspatch := fmt.Sprintf(`{"metadata":{"labels":{"testLabel":"testValue"}}}`, namespaceName)
```
Comment (neolit123) [inline test/e2e/apimachinery/namespace.go]:
```
nspatch := `{"metadata":{"labels":{"testLabel":"testValue"}}}`
```
should suffice here.

this is failing govet and staticcheck:
https://prow.k8s.io/view/gcs/kubernetes-jenkins/pr-logs/pull/86854/pull-kubernetes-verify/1213988224351866881

Comment (BobyMCbobs) [reply test/e2e/apimachinery/namespace.go]: Previously the patch was also including the name of the namespace, I removed that.

Updated in https://github.com/kubernetes/kubernetes/pull/86854/commits/519ad8771b25723e13ca42cdb005fd3d4a658102

File: test/e2e/apimachinery/namespace.go [hunk 8]
```diff
@@ -246,4 +248,29 @@ var _ = SIGDescribe("Namespaces [Serial]", func() {
...
+			ObjectMeta: metav1.ObjectMeta{
+				Name: namespaceName,
+			},
+		})
+		framework.ExpectNoError(err, "failed creating Namespace")
+
+		ginkgo.By("patching the Namespace")
+		nspatch := `{"metadata":{"labels":{"testLabel":"testValue"}}}`
+		_, err = f.ClientSet.CoreV1().Namespaces().Patch(namespaceName, types.StrategicMergePatchType, []byte(nspatch))
+		framework.ExpectNoError(err, "failed to patch Namespace")
```
Comment (johnSchnake) [inline test/e2e/apimachinery/namespace.go]: If it failed here, wouldn't we orphan the new ns we made? I'm not exactly certain though how the `Expect...` works w.r.t. defers; even if you did a 'defer deleteNS(...)` would the namespace still be orphaned on failure?

**Review by johnSchnake — APPROVED** [review]
/lgtm
/retest

Only nit was about if there is a chance we orphan a NS; but I don't know if there is a better way to do it with the way that control flow works with `Expect...` anyways.

**Comment by fejta** [comment]
Please create an OWNERS file in https://github.com/kubernetes/kubernetes/tree/1d906b259886cb44274f2239558640416a952a97/test/e2e/apimachinery

**Comment by BobyMCbobs** [comment]
> Please create an OWNERS file in https://github.com/kubernetes/kubernetes/tree/1d906b259886cb44274f2239558640416a952a97/test/e2e/apimachinery

OK, I've added one

**Comment by BobyMCbobs** [comment]
> /lgtm
> /retest
> 
> Only nit was about if there is a chance we orphan a NS; but I don't know if there is a better way to do it with the way that control flow works with `Expect...` anyways.

I've updated the test to include a `defer` statement to cleanup the created namespace as suggested.

File: test/e2e/apimachinery/namespace.go (hunk 8 above)
Comment (neolit123) [reply test/e2e/apimachinery/namespace.go]: @johnSchnake 
cleanup of failure (defer) is not a common pattern in this file and other tests.
the namespace will be orphaned, but essentially this is just a test, so it's fine.

@BobyMCbobs 
`defer deleteNamespace(f, namespaceName)` will now do a second delete that will always fail on success, if the previous code passes:
```
		ginkgo.By("deleting the Namespace")
		err = f.ClientSet.CoreV1().Namespaces().Delete(namespaceName, &metav1.DeleteOptions{})
		framework.ExpectNoError(err, "failed to delete the test Namespace")
```

i don't think we have to have the deferred delete.

File: test/e2e/apimachinery/OWNERS [hunk 9]
```diff
@@ -0,0 +1,8 @@
+# See the OWNERS docs at https://go.k8s.io/owners
+
+reviewers:
+  - sig-apimachinery-maintainers
```
Comment (neolit123) [inline test/e2e/apimachinery/OWNERS]: there is no such owner alias in:
https://github.com/kubernetes/kubernetes/blob/master/OWNERS_ALIASES

this line should be:
```
  - sig-api-machinery-api-reviewers
```

File: test/e2e/apimachinery/OWNERS [hunk 10]
```diff
@@ -0,0 +1,8 @@
...
+approvers:
+  - sig-apimachinery-maintainers
```
Comment (neolit123) [inline test/e2e/apimachinery/OWNERS]: this line should be:
```
  - sig-api-machinery-api-approvers
```

File: test/e2e/apimachinery/namespace.go (hunk 8 above)
Comment (BobyMCbobs) [reply test/e2e/apimachinery/namespace.go]: OK, I'll remove it.

File: test/e2e/apimachinery/OWNERS (hunk 9 above)
Comment (BobyMCbobs) [reply test/e2e/apimachinery/OWNERS]: Updated in https://github.com/kubernetes/kubernetes/pull/86854/commits/1fa8ddd8aa6e347fe6a6500f74dba6310dd68293

File: test/e2e/apimachinery/OWNERS (hunk 10 above)
Comment (BobyMCbobs) [reply test/e2e/apimachinery/OWNERS]: Updated in https://github.com/kubernetes/kubernetes/pull/86854/commits/1fa8ddd8aa6e347fe6a6500f74dba6310dd68293

File: test/e2e/apimachinery/OWNERS [hunk 11]
```diff
@@ -0,0 +1,8 @@
+# See the OWNERS docs at https://go.k8s.io/owners
+
+reviewers:
+  - sig-api-machinery-api-approvers
```
Comment (neolit123) [inline test/e2e/apimachinery/OWNERS]: i pointed [here](https://github.com/kubernetes/kubernetes/pull/86854#discussion_r363476386) this should be:
```
sig-api-machinery-api-reviewers
```

but i realized these are API reviewers / approvers and this owner alias will not work.

the OWNERS file should have individuals like this one:
https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apimachinery/OWNERS

but the problem is that we don't know how wants to participate.

so @BobyMCbobs please remove the OWNERS file from this PR / commits and we can log a separate issue.

Comment (neolit123) [reply test/e2e/apimachinery/OWNERS]: > so @BobyMCbobs please remove the OWNERS file from this PR / commits and we can log a separate issue.

done logged:
https://github.com/kubernetes/kubernetes/issues/86888

Comment (BobyMCbobs) [reply test/e2e/apimachinery/OWNERS]: Removed in aeed347ba88

**Comment by neolit123** [comment]
please squash your commits to one.
then LGTM.

**Comment by BobyMCbobs** [comment]
> please squash your commits to one.
> then LGTM.

I've squashed them in 9951a9c98e2

**Comment by neolit123** [comment]
/lgtm
/assign @oomichi 
for approval.

thanks.

File: test/e2e/apimachinery/namespace.go [hunk 12]
```diff
@@ -246,4 +249,30 @@ var _ = SIGDescribe("Namespaces [Serial]", func() {
...
 		func() { extinguish(f, 100, 0, 150) })
 
+	ginkgo.It("should patch a Namespace", func() {
+		ginkgo.By("creating a Namespace")
+		namespaceName := "nspatchtest-" + string(uuid.NewUUID())
+		_, err := f.ClientSet.CoreV1().Namespaces().Create(&v1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: namespaceName,
+			},
+		})
```
Comment (oomichi) [inline test/e2e/apimachinery/namespace.go]: Why don't you use `f.CreateNamespace()` like the other e2e tests instead of calling the raw API?

File: test/e2e/apimachinery/namespace.go [hunk 13]
```diff
@@ -246,4 +249,30 @@ var _ = SIGDescribe("Namespaces [Serial]", func() {
...
+
+		ginkgo.By("patching the Namespace")
+		nspatch := `{"metadata":{"labels":{"testLabel":"testValue"}}}`
+		_, err = f.ClientSet.CoreV1().Namespaces().Patch(namespaceName, types.StrategicMergePatchType, []byte(nspatch))
+		framework.ExpectNoError(err, "failed to patch Namespace")
+
+		ginkgo.By("get the Namespace and ensuring it has the label")
+		namespace, err := f.ClientSet.CoreV1().Namespaces().Get(namespaceName, metav1.GetOptions{})
+		framework.ExpectNoError(err, "failed to get Namespace")
+		gomega.Expect(namespace.ObjectMeta.Labels["testLabel"]).To(gomega.Equal("testValue"), "namespace not patched")
```
Comment (oomichi) [inline test/e2e/apimachinery/namespace.go]: We can use `framework.ExpectEqual(namespace.ObjectMeta.Labels["testLabel"], "testValue", "namespace not patched")` here instead of `gomega.Expect(...)`

File: test/e2e/apimachinery/namespace.go [hunk 14]
```diff
@@ -30,8 +30,11 @@ import (
 	"k8s.io/kubernetes/test/e2e/framework"
 	e2epod "k8s.io/kubernetes/test/e2e/framework/pod"
 	imageutils "k8s.io/kubernetes/test/utils/image"
+	"k8s.io/apimachinery/pkg/util/uuid"
 
 	"github.com/onsi/ginkgo"
+	"github.com/onsi/gomega"
+	"k8s.io/apimachinery/pkg/types"
```
Comment (oomichi) [inline test/e2e/apimachinery/namespace.go]: This import line should be moved to line 33 before `"k8s.io/apimachinery/pkg/util/uuid"`

**Review by oomichi — COMMENTED** [review]
Most part seems good for me, just few comments.

File: test/e2e/apimachinery/namespace.go (hunk 12 above)
Comment (BobyMCbobs) [reply test/e2e/apimachinery/namespace.go]: Updated in bd6550fb349

File: test/e2e/apimachinery/namespace.go (hunk 13 above)
Comment (BobyMCbobs) [reply test/e2e/apimachinery/namespace.go]: Updated in bd6550fb349

File: test/e2e/apimachinery/namespace.go (hunk 14 above)
Comment (BobyMCbobs) [reply test/e2e/apimachinery/namespace.go]: Updated in bd6550fb349

File: test/e2e/apimachinery/namespace.go [hunk 15]
```diff
@@ -246,4 +248,22 @@ var _ = SIGDescribe("Namespaces [Serial]", func() {
 	ginkgo.It("should always delete fast (ALL of 100 namespaces in 150 seconds) [Feature:ComprehensiveNamespaceDraining]",
 		func() { extinguish(f, 100, 0, 150) })
 
+	ginkgo.It("should patch a Namespace", func() {
+		ginkgo.By("creating a Namespace")
+		namespaceName := "nspatchtest-" + string(uuid.NewUUID())
+		ns, err := f.CreateNamespace(namespaceName, nil)
+		namespaceName = ns.ObjectMeta.Name
```
Comment (oomichi) [inline test/e2e/apimachinery/namespace.go]: If an error happens at the above `CreateNamespace()` call, `ns` will be nil and this line 255 makes an exception.
It is better to check `err` like line 256 before the above line.

Comment (BobyMCbobs) [reply test/e2e/apimachinery/namespace.go]: Good observation, updated in 2754cc37ea5

**Comment by fejta** [comment]
I still don't see an OWNERS file in https://github.com/kubernetes/kubernetes/tree/2754cc37ea569ce9478b9ffb1304cd9feeb1280e/test/e2e/apimachinery

Did you forget to add it to the commit?

**Comment by BobyMCbobs** [comment]
> I still don't see an OWNERS file in https://github.com/kubernetes/kubernetes/tree/2754cc37ea569ce9478b9ffb1304cd9feeb1280e/test/e2e/apimachinery
> 
> Did you forget to add it to the commit?

@fejta, I removed it from this PR per [the comments](https://github.com/kubernetes/kubernetes/pull/86854#discussion_r363481400) of @neolit123:

> the OWNERS file should have individuals like this one:
> https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apimachinery/OWNERS
> 
> but the problem is that we don't know who wants to participate.
> 
> so @BobyMCbobs please remove the OWNERS file from this PR / commits and we can log a separate issue.

TLDR: the addition of this file belongs in another PR

**Comment by oomichi** [comment]
The code itself seems good for me, thanks for updating.

/lgtm

**Comment by fejta** [comment]
I would either like to see the OWNERS file in this PR (in which case I will approve it) or else first resolve https://github.com/kubernetes/kubernetes/issues/86888 (in which case apimachinery or I can approve it)

**Comment by BobyMCbobs** [comment]
> I would either like to see the OWNERS file in this PR (in which case I will approve it) or else first resolve #86888 (in which case apimachinery or I can approve it)

@fejta, I've copied the contents of the staging version in 616003a8e96

File: test/e2e/apimachinery/OWNERS [hunk 16]
```diff
@@ -0,0 +1,27 @@
...
+- cheftako
+- mikedanese
+- liggitt
+- gmarek
+- sttts
+- ncdc
+- logicalhan
+- tallclair
+labels:
+- sig/api-machinery
```
Comment (oomichi) [inline test/e2e/apimachinery/OWNERS]: `No newlkine at end of file` is marked, is that fine?

**Comment by BobyMCbobs** [comment]
I've updated this test to comply with what was recommended by @liggitt on #86857.

File: test/e2e/apimachinery/namespace.go [hunk 17]
```diff
@@ -21,17 +21,20 @@ import (
 	"strings"
 	"sync"
 	"time"
+	"encoding/json"
```
Comment (oomichi) [inline test/e2e/apimachinery/namespace.go]: need to put this line(`"encoding/json"`) to line 20 before `"fmt"` to pass gofmt check.

File: test/e2e/apimachinery/namespace.go [hunk 18]
```diff
@@ -246,4 +249,26 @@ var _ = SIGDescribe("Namespaces [Serial]", func() {
...
+		ginkgo.By("creating a Namespace")
+		namespaceName := "nspatchtest-" + string(uuid.NewUUID())
+		ns, err := f.CreateNamespace(namespaceName, nil)
+		framework.ExpectNoError(err, "failed creating Namespace")
+		namespaceName = ns.ObjectMeta.Name
+
+		ginkgo.By("patching the Namespace")
+		nspatch, err := json.Marshal(map[string]interface{}{
+			"metadata": map[string]interface{}{
+				"labels": map[string]string{"testLabel":"testValue"},
```
Comment (oomichi) [inline test/e2e/apimachinery/namespace.go]: ditto
```
-				"labels": map[string]string{"testLabel":"testValue"},
+				"labels": map[string]string{"testLabel": "testValue"},
```

File: test/e2e/apimachinery/namespace.go [hunk 19]
```diff
@@ -246,4 +249,26 @@ var _ = SIGDescribe("Namespaces [Serial]", func() {
...
+			},
+		})
```
Comment (oomichi) [inline test/e2e/apimachinery/namespace.go]: need to add error check like `framework.ExpectNoError(err)` to pass the following staticcheck:
```
Errors from staticcheck:
test/e2e/apimachinery/namespace.go:260:12: this value of err is never used (SA4006)
```

File: test/e2e/apimachinery/namespace.go (hunk 17 above)
Comment (BobyMCbobs) [reply test/e2e/apimachinery/namespace.go]: Updated in 9ed55bc5203

File: test/e2e/apimachinery/namespace.go (hunk 18 above)
Comment (BobyMCbobs) [reply test/e2e/apimachinery/namespace.go]: Updated in 9ed55bc5203

File: test/e2e/apimachinery/namespace.go (hunk 19 above)
Comment (BobyMCbobs) [reply test/e2e/apimachinery/namespace.go]: Updated in 9ed55bc5203
